actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -123,9 +123,15 @@ def assert_enqueued_emails(number, &block)
     #       ContactMailer.with(email: 'user@example.com').welcome.deliver_later
     #     end
     #   end
-    def assert_enqueued_email_with(mailer, method, args: nil, queue: ActionMailer::Base.deliver_later_queue_name || "default", &block)
+    def assert_enqueued_email_with(mailer, method, params: nil, args: nil, queue: ActionMailer::Base.deliver_later_queue_name || "default", &block)
+      if mailer.is_a? ActionMailer::Parameterized::Mailer
```

Comment:
This is ugly, but sure makes the assertion nice! This allows the test code to use mailer.with just like the code being tested does.

Reply:
This is a nice extra bit of flexibility! They may have a policy against code smells such as using `instance_variable_get`, but I do see it used in many places throughout the Rails code base. The fact that it is tested, and the instance variable is within the same module, will likely improve the chances that they will allow it because if the variable is renamed or the implementation changes it will break the test within the same gem.

Reply:
Yeah, this is extremely flexible!